### PR TITLE
update to 1.15-pre3

### PR DESCRIPTION
**Pull request by TwilightFlower**
1.15 is on the doorstep. Time to update.

Major changes:
*Lots of renames.
*Temporarily disabled config functionality, as TinyConfig has not updated.
*Temporarily disabled trinkets functionality, as Trinkets has not updated.
*1.15 rendering changes. Lots of them.

Shotgun projectile rendering is broken, but the shotgun has two outstanding crash bugs so I considered it low priority.

### CODE REVIEW

File: src/main/java/com/fabriccommunity/thehallow/TheHallowClient.java
Comment (LemmaEOF) [inline src/main/java/com/fabriccommunity/thehallow/TheHallowClient.java]: Do we actually want fluids to be on the solid render layer? I'd think we'd want them to be see-through a bit...

File: src/main/java/com/fabriccommunity/thehallow/client/model/CultistModel.java
Comment (LemmaEOF) [inline src/main/java/com/fabriccommunity/thehallow/client/model/CultistModel.java:70]: Does this render everything properly right now?

File: src/main/java/com/fabriccommunity/thehallow/client/render/PumpcownStemFeatureRenderer.java
Comment (LemmaEOF) [inline src/main/java/com/fabriccommunity/thehallow/client/render/PumpcownStemFeatureRenderer.java]: Is this just GitHub making the indentation look weird? If not, it should be bumped back one.

File: src/main/java/com/fabriccommunity/thehallow/item/HallowCharmItem.java
Comment (LemmaEOF) [inline src/main/java/com/fabriccommunity/thehallow/item/HallowCharmItem.java]: Add a TODO here so we don't forget it?

File: src/main/java/com/fabriccommunity/thehallow/item/PaperBagItem.java
Comment (LemmaEOF) [inline src/main/java/com/fabriccommunity/thehallow/item/PaperBagItem.java]: Add a TODO here so we don't forget it?

File: src/main/java/com/fabriccommunity/thehallow/item/PumpkinRing.java
Comment (LemmaEOF) [inline src/main/java/com/fabriccommunity/thehallow/item/PumpkinRing.java]: Add a TODO here so we don't forget it?

File: src/main/java/com/fabriccommunity/thehallow/item/SkirtCostume.java
Comment (LemmaEOF) [inline src/main/java/com/fabriccommunity/thehallow/item/SkirtCostume.java]: Add a TODO here so we don't forget it?

File: src/main/java/com/fabriccommunity/thehallow/mixin/EndermanEntityMixin.java
Comment (LemmaEOF) [inline src/main/java/com/fabriccommunity/thehallow/mixin/EndermanEntityMixin.java]: Add a TODO here so we don't forget it?

File: src/main/java/com/fabriccommunity/thehallow/mixin/PlayerEntityMixin.java
Comment (LemmaEOF) [inline src/main/java/com/fabriccommunity/thehallow/mixin/PlayerEntityMixin.java]: Add a TODO here so we don't forget it?

File: src/main/java/com/fabriccommunity/thehallow/mixin/ServerPlayNetworkHandlerMixin.java
Comment (LemmaEOF) [inline src/main/java/com/fabriccommunity/thehallow/mixin/ServerPlayNetworkHandlerMixin.java]: We don't need to comment out the tag check, do we? That can probably stay in for now.

File: src/main/java/com/fabriccommunity/thehallow/registry/HallowedItems.java
Comment (LemmaEOF) [inline src/main/java/com/fabriccommunity/thehallow/registry/HallowedItems.java]: Add a TODO here so we don't forget it?

**Review by LemmaEOF — APPROVED** [review]
Looking pretty good to me! Let's not merge this till TinyConfig and Trinkets are updated, though.

File: src/main/java/com/fabriccommunity/thehallow/TheHallowClient.java
Comment (TwilightFlower) [reply src/main/java/com/fabriccommunity/thehallow/TheHallowClient.java]: This did stick out to me as odd, but this is how they are in 1.14 so I kept it.

Comment (LemmaEOF) [reply src/main/java/com/fabriccommunity/thehallow/TheHallowClient.java]: I think it's ok for now then. I'll take a look at some point.

**Review by LemmaEOF — APPROVED** [review]
Looking good! This should be mergeable once deps update!

**Comment by LemmaEOF** [comment]
@Nuclearfarts Trinkets and TinyConfig are now both updated. Are you able to update them here or should I do it after merge?

**Comment by LemmaEOF** [comment]
Awesome! Don't forget to fix all the TODO-marked places we have trinkets stuff commented out!

**Comment by TwilightFlower** [comment]
Alright, back at my normal PC instead of my laptop that Gradle dislikes. Uncommented the trinkets stuff, but I can't do the rendering quite yet as trinkets doesn't pass on the MatrixStack or VertexConsumerProvider.

**Comment by LemmaEOF** [comment]
Ah, that’s weird. Thought it did.

**Comment by TwilightFlower** [comment]
alright then. Trinkets rendering has been fixed, though I had to write my own version of a trinkets utility method as it still used GlStateManager. Even though the tinyconfig github does show a 1.15 version, there doesn't seem to be a build anywhere, so I switched to just using jankson. (I thought the config worked before, it didn't, I'd forgotten to uncomment it.) I found out that the option to disable pumpkins on mobs' heads never worked while testing this and fixed that as well.

File: src/main/java/com/fabriccommunity/thehallow/TheHallowClient.java
Comment (Juuxel) [inline src/main/java/com/fabriccommunity/thehallow/TheHallowClient.java]: You can use `putBlocks` here instead of calling `putBlock` for each block.

File: src/main/java/com/fabriccommunity/thehallow/TheHallowClient.java
Comment (Juuxel) [inline src/main/java/com/fabriccommunity/thehallow/TheHallowClient.java]: You can use `putFluids` here instead of calling `putFluid` for each fluid.

**Review by Juuxel — CHANGES_REQUESTED** [review]

**Review by LemmaEOF — APPROVED** [review]
Alright, I think this leaves us merge ready! I'll put this on last call.

File: gradle.properties
Comment (LemmaEOF) [inline gradle.properties:39]: I'll bump this up to LibCD 2.0 in a separate PR once this is merged.

File: src/main/java/com/fabriccommunity/thehallow/client/render/HallowedTreasureChestBlockEntityRenderer.java
Comment (vini2003) [inline src/main/java/com/fabriccommunity/thehallow/client/render/HallowedTreasureChestBlockEntityRenderer.java:44]: holy shit someone left comments

**Review by vini2003 — APPROVED** [review]
Everything seems fine, way too many changes to check one by one but I trust you guys.

**Review by ImAK9 — APPROVED** [review]
looks good to me!

**Pull request merged**
